checks/checkforcontent.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2020 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checks
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+	"strings"
+
+	"github.com/ossf/scorecard/checker"
+)
+
+// CheckFilesContent downloads the tar of the repository and calls the onFileContent() function
```

Comment:
A `func` within another `func` is little confusing 
func(path string, content []byte,
	Logf func(s string, f ...interface{})) bool)

Reply:
I followed what's done for CheckIfFileExists. Is there another way to pass the log function to the callback?